pinot-common/src/main/java/org/apache/pinot/common/utils/TarCompressionUtils.java

```diff
@@ -94,14 +107,27 @@ public static void createCompressedTarFile(File inputFile, File outputFile)
   public static void createCompressedTarFile(File[] inputFiles, File outputFile)
       throws IOException {
     String compressorName = null;
-    for (String supportedCompressorExtension : COMPRESSOR_NAME_BY_FILE_EXTENSIONS.keySet()) {
-      if (outputFile.getName().endsWith(supportedCompressorExtension)) {
-        compressorName = COMPRESSOR_NAME_BY_FILE_EXTENSIONS.get(supportedCompressorExtension);
-        break;
+    if (outputFile.getName().endsWith(TAR_COMPRESSED_FILE_EXTENSION)) {
+      createCompressedTarFile(inputFiles, outputFile, _defaultCompressorName);
+    } else {
+      for (String supportedCompressorExtension : COMPRESSOR_NAME_BY_FILE_EXTENSIONS.keySet()) {
+        if (outputFile.getName().endsWith(supportedCompressorExtension)) {
+          compressorName = COMPRESSOR_NAME_BY_FILE_EXTENSIONS.get(supportedCompressorExtension);
+          createCompressedTarFile(inputFiles, outputFile, compressorName);
```

Comment:
we can move common code createCompressedTarFile(inputFiles, outputFile, compressorName) after precondition check.